metadata_layer.py

```diff
@@ -898,67 +941,73 @@ def apply(self, pool, replace=None):
             )
             new_lpdms.append(lpdm)
 
         if replace.formats:
+            # If any LicensePoolDeliveryMechanisms were not mentioned in
+            # self.formats, remove them.
             for lpdm in old_lpdms:
                 if lpdm not in new_lpdms:
                     for loan in lpdm.fulfills:
                         self.log.info("Loan %i is associated with a format that is no longer available. Deleting its delivery mechanism." % loan.id)
                         loan.fulfillment = None
                     _db.delete(lpdm)
 
-        changed_licenses = False
-        if pool:
-            # if we were not passed a last_checked value, then update pool as of now
-            if not self.last_checked: 
-                # Update availabily information. This may result in the issuance
-                # of additional events.
-                changed_licenses = pool.update_availability(
-                    new_licenses_owned=self.licenses_owned,
-                    new_licenses_available=self.licenses_available,
-                    new_licenses_reserved=self.licenses_reserved,
-                    new_patrons_in_hold_queue=self.patrons_in_hold_queue,
-                    as_of=datetime.datetime.utcnow(),
-                )
-            else:
-                # if we were passed a last_checked value, such as when creating the 
-                # CirculationData object in the bibliographic parser for a 3m log, then 
-                # check if that log update value is after the license pool's current knowledge, 
-                # and only update license infos if new information is really new.
-                if (pool.last_checked and (self.last_checked >= pool.last_checked)):
-                    # Update availabily information. This may result in the issuance
-                    # of additional events.
-                    changed_licenses = pool.update_availability(
-                        new_licenses_owned=self.licenses_owned,
-                        new_licenses_available=self.licenses_available,
-                        new_licenses_reserved=self.licenses_reserved,
-                        new_patrons_in_hold_queue=self.patrons_in_hold_queue,
-                        as_of=self.last_checked
-                    )
-
-        # Changes to the delivery mechanisms may have changed the work's
-        # open-access status.
-        old_open_access = pool.open_access                    
-        for lpdm in pool.delivery_mechanisms:
+        # Changes to the delivery mechanisms may have changed the
+        # open-access status of all LicensePools for this work.
+        pools = identifier.licensed_through
+        old_open_access = any(pool.open_access for pool in pools)
+        for lpdm in identifier.delivery_mechanisms:
             if (lpdm.rights_status
                 and lpdm.rights_status.uri in RightsStatus.OPEN_ACCESS):
-                pool.open_access = True
+                new_open_access = True
```

Comment:
We need it in case a book _stops_ being open access because a LPDM was _removed_. I'll move that logic into model.py though.